### ISSUE TITLE
[Feature Request] It should be possible to create el-menu-item without route

### ISSUE THREAD

**Issue by georgyfarniev** [ElemeFE/element#13292]
### Existing Component
Yes

### Component Name
NavMenu

### Description
Currently, there's no straightforward way to create el-menu-item which will not affect current route in menu.

Good use case: we need to open modal dialog when click menu item WITHOUT changing if current route.

Proposed solution:  allow to omit index parameter of el-menu-item or set it to null, in this case clicking on this item will NOT affect current route. 

I think this is very common use case for navigation menu and element should definitely cover it. 

Please let me know if this use case is accepted for inclusion in el-menu-item functionality and I will try to make PR by myself, as this feature is very important for me. Thank you.

**Comment by csvwolf** [comment]
https://codepen.io/csvwolf/pen/jQEYrz Do you mean you want this effect?

**Comment by georgyfarniev** [comment]
> https://codepen.io/csvwolf/pen/jQEYrz Do you mean you want this effect?

No, I want to add menu item which will not affect routes and current selected menu item at all. For example: 

we have navigation menu with some of el-menu-item's should NOT have any index value and should NOT change current route. 

More detailed use case:
Let's say, user stay at page `profile`, then he clicks **Contact support** link on el-menu. This action will open modal dialog, but most important that user should stay at current page and the selected element of menu shouldn't be changed. In other words, it would be nice to have Non-selectable and non-routable el-menu-item's on el-menu.

Current properties of el-menu-item enforce to specify index, and even if it is null, it's not working as expected.

Please let me know if my explanation is not clear, I will record video if necessary.

**Comment by csvwolf** [comment]
@georgyfarniev Emmm, you mean: you want to change the native behavior for not changing index in the menu? Since I think my demo is not changing page.

You just want something clickable, it's in the menu, but not the part of menu?

**Comment by georgyfarniev** [comment]
@csvwolf yes, it should be in menu and styled as menu element, but should not affect route state and menu index at all. Just only accept click to perform some action. I did little dig into the code and I found way to do it in the following way:

1) Allow null value and set index for el-menu-item to null by default
2) Take a look at packages/menu/src/menu.vue:251: should be two checks: 

```diff
++ const hasIndex = item.index !== null
-- this.activeIndex = item.index;
++ if (hasIndex) {
++   this.activeIndex = item.index;
++ }
-- if (this.router) {
++ if (this.router && hasIndex) {
```

I didn't test this code thoroughly, but it worked for me. Currently, i'm trying to fix node-gyp build for element on my windows 10 machine to make appropriate changes (changes above tried only in already compiled file) and prepare appropriate pull-request for this feature.
